### PR TITLE
Initial draft of LoC mapping

### PR DESCRIPTION
**Pull request by moltude**

### CODE REVIEW

**Comment by moltude** [comment]
@mdellabitta implemented your suggestions

File: lib/mappers/library_of_congress_mapper.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        if languages:
+            self.update_source_resource({"language": languages})
+
+    def map_spatial(self):
+        places = set()
+        values = iterify(getprop(self.provider_data, "location", True))
+        for v in values:
+            places = places.add(v.keys())
+        if places:
+            self.update_source_resource({"spatial": places})
```
Comment (mdellabitta) [inline lib/mappers/library_of_congress_mapper.py]: Does json.dumps help you out with an implicit conversion to a list here?

File: lib/mappers/library_of_congress_mapper.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            self.update_source_resource({"contributor": contributors})
+
+    def map_description(self):
+        props = ['description', 'created_published']
+        description = []
+
+        for prop in props:
+            values = iterify(getprop(self.provider_data, prop, True))
+            if values:
+                description = list(set(values+description))
```
Comment (mdellabitta) [inline lib/mappers/library_of_congress_mapper.py]: Another place you're converting back and forth from sets

File: lib/mappers/library_of_congress_mapper.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        if description:
+            self.update_source_resource({"description": description})
+
+    def map_format_genre_type(self, dest):
+        formats = []
+
+        # Either-Or mapping
+        values = iterify(getprop(self.provider_data, "type", True))
+        if values:
+            formats = list(set(values + formats))
```
Comment (mdellabitta) [inline lib/mappers/library_of_congress_mapper.py]: Here too

File: lib/mappers/library_of_congress_mapper.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        else:
+            values = iterify(getprop(self.provider_data, "format", True))
+            formats = values.keys()
+
+        # Include mapping from 'genre' if the destination is not 'type'
+        if dest != "type":
+            values = iterify(getprop(self.provider_data, "genre", True))
+            if values:
+                formats = list(set(values + formats))
```
Comment (mdellabitta) [inline lib/mappers/library_of_congress_mapper.py]: Here

File: lib/mappers/library_of_congress_mapper.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+            "medium": "extent"
+        }
+        for (sources, dest) in maps.iteritems():
+            for source in sources:
+                values = iterify(getprop(self.provider_data, source, True))
+                if values:
+                    existing_values = \
+                        getprop(self.mapped_data["sourceResource"], dest, True)
+                    if existing_values:
+                        values = list(set(values + existing_values))
```
Comment (mdellabitta) [inline lib/mappers/library_of_congress_mapper.py]: here

**Comment by moltude** [comment]
Closing PR until we address some of the big picture issues at play here.

**Pull request closed without merging**